test/system/src/test/java/io/pravega/test/system/MultiControllerTest.java

```diff
@@ -144,11 +142,13 @@ public void multiControllerTest() throws ExecutionException, InterruptedExceptio
         AssertExtensions.assertThrows("Should throw RetriesExhaustedException",
                 createScope("scope" + RandomStringUtils.randomAlphanumeric(10), controllerURIDirect.get()),
                 throwable -> throwable instanceof RetriesExhaustedException);
+        controllerClient.close();
```

Comment:
Instead of doing this, could you create the controller client instance in the setup method and use the `@Cleanup` annotation? Alternatively, you could create the controller client instance in this method and pass it to `createScope` (which is a private method).

Reply:
As per the comment here: https://github.com/pravega/pravega/pull/2542#discussion_r186336592
I need to create new `controller client` instance every time I create a scope.  Also, I have tried the ways you have mentioned above, but it is trying to access any available controller URI in the case of `pravega://` and scope gets created, in case of 0 controller instances.

Reply:
This is problematic as we have to remember to close the controller client every time. In the future, this makes maintenance of the code more difficult. Not to say that there is the risk that `controllerClient` is null and we end up with NPE.

I think you can refactor `createScope` to make it work.